code/include/playfab_c/PlayFabClientApi_c.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include "httpClient/pal.h"
+#include "httpClient/async.h"
+#include "PlayFabClientDataModels_c.h"
+
+/// <summary>
+/// Initialize PlayFab global state
+/// </summary>
+HRESULT PlayFabInitialize(
+    _In_ const char* titleId
+) noexcept;
+
+/// <summary>
+/// Cleanup PlayFab global state. Shouldn't be called while async APIs are still running.
+/// </summary>
+HRESULT PlayFabCleanupAsync(
+    _In_ XAsyncBlock* async
+) noexcept;
+
+/// <summary>
+/// Get the result handle for an asynchronous API
+/// </summary>
+HRESULT PlayFabGetResultHandle(
+    _In_ XAsyncBlock* async,
+    _Out_ PlayFabResultHandle* resultHandle
+) noexcept;
+
+/// <summary>
+/// Duplicate result.
+/// </summary>
+HRESULT PlayFabResultDuplicateHandle(
+    _In_ PlayFabResultHandle resultHandle,
+    _Out_ PlayFabResultHandle* duplicatedHandle
+) noexcept;
+
+/// <summary>
+/// Close result. Invalidates all pointers associated with the result handle.
+/// </summary>
+void PlayFabResultCloseHandle(
+    _In_ PlayFabResultHandle resultHandle
+) noexcept;
+
+/// <summary>
+/// Login with custom ID
+/// </summary>
+HRESULT PlayFabLoginWithCustomIDAsync(
+    _In_ const PlayFabLoginWithCustomIDRequest* request,
+    _In_ XAsyncBlock* async
+) noexcept;
+
+/// <summary>
+/// Retrieves PlayFabUserHandle from a login result. User handle is needed to make future API calls
+/// </summary>
+HRESULT PlayFabLoginWithCustomIDResultGetUserHandle(
+    _In_ PlayFabResultHandle resultHandle,
+    _Out_ PlayFabUserHandle* userHandle
+) noexcept;

```

Comment:
why make them call

HRESULT PlayFabGetResultHandle(
    _In_ XAsyncBlock* async,
    _Out_ PlayFabResultHandle* resultHandle
) noexcept;

then 

HRESULT PlayFabLoginWithCustomIDResultGetUserHandle(
    _In_ PlayFabResultHandle resultHandle,
    _Out_ PlayFabUserHandle* userHandle
) noexcept;

when you could expose a single API like so:

HRESULT PlayFabLoginWithCustomIDResult(
    _In_ XAsyncBlock* async,
    _Out_ PlayFabUserHandle* userHandle
) noexcept;

Reply:
I've not exposed APIs to retrieve them yet, but there are things other than the userHandle returned from a login call (for example, last login time, is the account newly created, etc.).  Some of the returned info didn't feel appropriate as part of the User object so I was going to just return it separately from the login call. 

That said, I agree this feels a little bit clunky and I might change it.  I am obviously still open to suggestions as well.

---
In reply to: [590781842](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r590781842) [](ancestors = 590781842)